hardware/1Wire.cpp

```diff
@@ -509,83 +481,33 @@ void C1Wire::ReportTemperatureHumidity(const std::string& deviceId, const float
+        uint16_t lID = (deviceIdByteArray[0] << 8) | deviceIdByteArray[1];
+	SendTempHumSensor(lID, 255, temperature, round(humidity), "TempHum");
 }
 
 void C1Wire::ReportLightState(const std::string& deviceId, const int unit, const bool state)
 {
-#if defined(_DEBUG)
-	_log.Log(LOG_STATUS, "device '%s' unit %d state is %s", deviceId.c_str(), unit, (state) ? "on" : "off");
-#endif
-
-// check - is state changed ?
-	char num[16];
-	sprintf(num, "%s/%d", deviceId.c_str(), unit);
-	const std::string id(num);
-
-	std::map<std::string, bool>::iterator it;
-	it = m_LastSwitchState.find(id);
-	if (it != m_LastSwitchState.end())
-	{
-		if (m_LastSwitchState[id] == state)
-		{
-			return;
-		}
-	}
-
-#if defined(_DEBUG)
-	_log.Log(LOG_STATUS, "device '%s' unit %d changed state to %s", deviceId.c_str(), unit, (state) ? "on" : "off");
-#endif
-
-	m_LastSwitchState[id] = state;
-
 	unsigned char deviceIdByteArray[DEVICE_ID_SIZE]={0};
 	DeviceIdToByteArray(deviceId,deviceIdByteArray);
+	int lID = (deviceIdByteArray[0] << 24) + (deviceIdByteArray[1] << 16) + (deviceIdByteArray[2] << 8) + deviceIdByteArray[3];
 
-	RBUF tsen;
-	memset(&tsen,0,sizeof(RBUF));
-	tsen.LIGHTING2.packetlength=sizeof(tsen.LIGHTING2)-1;
-	tsen.LIGHTING2.packettype=pTypeLighting2;
-	tsen.LIGHTING2.subtype=sTypeAC;
-	tsen.LIGHTING2.seqnbr=0;
-	tsen.LIGHTING2.id1=(BYTE)deviceIdByteArray[0];
-	tsen.LIGHTING2.id2=(BYTE)deviceIdByteArray[1];
-	tsen.LIGHTING2.id3=(BYTE)deviceIdByteArray[2];
-	tsen.LIGHTING2.id4=(BYTE)deviceIdByteArray[3];
-	tsen.LIGHTING2.unitcode=unit;
-	tsen.LIGHTING2.cmnd=state?light2_sOn:light2_sOff;
-	tsen.LIGHTING2.level=0;
-	tsen.LIGHTING2.rssi=12;
-	sDecodeRXMessage(this, (const unsigned char *)&tsen.LIGHTING2, NULL, 255);
+	SendSwitch(lID, unit, 255, state, 0, "Switch");
 }
 
 void C1Wire::ReportCounter(const std::string& deviceId, const int unit, const unsigned long counter)
 {
 	unsigned char deviceIdByteArray[DEVICE_ID_SIZE]={0};
 	DeviceIdToByteArray(deviceId,deviceIdByteArray);
 
-	RBUF tsen;
-	memset(&tsen,0,sizeof(RBUF));
-	tsen.RFXMETER.packetlength=sizeof(tsen.RFXMETER)-1;
-	tsen.RFXMETER.packettype=pTypeRFXMeter;
-	tsen.RFXMETER.subtype=sTypeRFXMeterCount;
-	tsen.RFXMETER.rssi=12;
-	tsen.RFXMETER.id1=(BYTE)deviceIdByteArray[0];
-	tsen.RFXMETER.id2=(BYTE)deviceIdByteArray[1] + unit;
-
-	tsen.RFXMETER.count1 = (BYTE)((counter & 0xFF000000) >> 24);
-	tsen.RFXMETER.count2 = (BYTE)((counter & 0x00FF0000) >> 16);
-	tsen.RFXMETER.count3 = (BYTE)((counter & 0x0000FF00) >> 8);
-	tsen.RFXMETER.count4 = (BYTE)(counter & 0x000000FF);
-	sDecodeRXMessage(this, (const unsigned char *)&tsen.RFXMETER, NULL, 255);
+	SendMeterSensor(deviceIdByteArray[0], deviceIdByteArray[1]+unit, 255, (const float)counter, "Counter");
 }
 
 void C1Wire::ReportVoltage(const std::string& deviceId, const int unit, const int voltage)
 {
 	if (voltage == -1000.0)
 		return;
 
+	// There is no matching SendXXX() function for this?
```

Comment:
We have a merge problem, is it possible to solve this if this patch needs to be merged?